`make` now logs uselessly verbose messages on compile errors

#122871 appears to have been trying to improve the logging in e2e test runs, but it seems to have affected much more than that, and now when a build fails, it logs so many useless messages that you have to scroll back to see why your compile actually failed.

before:
```none
> make WHAT=cmd/kube-proxy
go version go1.22rc2 linux/amd64
+++ [0205 09:33:29] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kube-proxy (static)
# k8s.io/kubernetes/cmd/kube-proxy/app
cmd/kube-proxy/app/server.go:723:3: syntax error: unexpected }, expected expression
!!! [0205 09:33:31] Call tree:
!!! [0205 09:33:31]  1: /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:788 kube::golang::build_some_binaries(...)
!!! [0205 09:33:31]  2: /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:949 kube::golang::build_binaries_for_platform(...)
!!! [0205 09:33:31]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0205 09:33:31] Call tree:
!!! [0205 09:33:31]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0205 09:33:31] Call tree:
!!! [0205 09:33:31]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make: *** [Makefile:96: all] Error 1
```

after:
```none
> make WHAT=cmd/kube-proxy
go version go1.22rc2 linux/amd64
+++ [0205 09:34:05] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kube-proxy (static)
# k8s.io/kubernetes/cmd/kube-proxy/app
cmd/kube-proxy/app/server.go:723:3: syntax error: unexpected }, expected expression
!!! [0205 09:34:05] Call tree:
!!! [0205 09:34:05]  1: /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:788 kube::golang::build_some_binaries(...)
!!! [0205 09:34:05]  2: /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:949 kube::golang::build_binaries_for_platform(...)
!!! [0205 09:34:05]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0205 09:34:05] Error in /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:747. 'GO111MODULE=on GOPROXY=off go install "${build_args[@]}" "$@"' exited with status 1
!!! [0205 09:34:05] Call tree:
!!! [0205 09:34:05]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0205 09:34:05] Error in /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:950. '( kube::golang::set_platform_envs "${platform}"; kube::golang::build_binaries_for_platform "${platform}" )' exited with status 1
!!! [0205 09:34:05] Call tree:
!!! [0205 09:34:05]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0205 09:34:05] Error in /home/danw/src/k8s.io/kubernetes/hack/lib/golang.sh:953. '( kube::golang::setup_env; V=2 kube::log::info "Go version: $(GOFLAGS='' go version)"; local host_platform; host_platform=$(kube::golang::host_platform); local goflags goldflags gogcflags gotags; goflags=(); gogcflags="${GOGCFLAGS:-}"; goldflags="all=$(kube::version::ldflags) ${GOLDFLAGS:-}"; if [[ "${DBG:-}" == 1 ]]; then
    gogcflags="${gogcflags} all=-N -l";
else
    goldflags="${goldflags} -s -w"; goflags+=("-trimpath");
fi; gotags="selinux,notest,$(echo "${GOFLAGS:-}" | sed -ne 's|.*-tags=\([^-]*\).*|\1|p')"; local -a targets=(); local arg; for arg in "$@";
do
    if [[ "${arg}" == -* ]]; then
        goflags+=("${arg}");
    else
        targets+=("${arg}");
    fi;
done; if [[ ${#targets[@]} -eq 0 ]]; then
    targets=("${KUBE_ALL_TARGETS[@]}");
fi; local -a platforms; IFS=" " read -ra platforms <<< "${KUBE_BUILD_PLATFORMS:-}"; if [[ ${#platforms[@]} -eq 0 ]]; then
    platforms=("${host_platform}");
fi; local -a binaries; while IFS="" read -r binary; do
    binaries+=("$binary");
done < <(kube::golang::binaries_from_targets "${targets[@]}"); local parallel=false; if [[ ${#platforms[@]} -gt 1 ]]; then
    local gigs; gigs=$(kube::golang::get_physmem); if [[ ${gigs} -ge ${KUBE_PARALLEL_BUILD_MEMORY} ]]; then
        kube::log::status "Multiple platforms requested and available ${gigs}G >= threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in parallel"; parallel=true;
    else
        kube::log::status "Multiple platforms requested, but available ${gigs}G < threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in serial"; parallel=false;
    fi;
fi; if [[ "${parallel}" == "true" ]]; then
    kube::log::status "Building go targets for {${platforms[*]}} in parallel (output will appear in a burst when complete):" "${targets[@]}"; local platform; for platform in "${platforms[@]}";
    do
        ( kube::golang::set_platform_envs "${platform}"; kube::log::status "${platform}: build started"; kube::golang::build_binaries_for_platform "${platform}"; kube::log::status "${platform}: build finished" ) &> "/tmp//${platform//\//_}.build" &
    done; local fails=0; for job in $(jobs -p);
    do
        wait "${job}" || (( fails+=1 ));
    done; for platform in "${platforms[@]}";
    do
        cat "/tmp//${platform//\//_}.build";
    done; exit "${fails}";
else
    for platform in "${platforms[@]}";
    do
        kube::log::status "Building go targets for ${platform}"; ( kube::golang::set_platform_envs "${platform}"; kube::golang::build_binaries_for_platform "${platform}" );
    done;
fi )' exited with status 1
make: *** [Makefile:96: all] Error 1
```

/assign @oxxenix @liggitt 
cc @pohly (re https://kubernetes.slack.com/archives/CHGFYJVAN/p1706868117299359)

@pohly It shows a little bit more location info with the change.
Here is how `make WHAT=cmd/kubelet` failure looks like with the change (but without BASH_COMMAND:
```
$ make WHAT=cmd/kubelet
go version go1.22rc2 linux/amd64
+++ [0206 15:57:18] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubelet (non-static)
# k8s.io/kubernetes/pkg/kubelet/cm/dra
pkg/kubelet/cm/dra/manager.go:67:53: syntax error: unexpected error at end of statement
!!! [0206 15:57:19] Call tree:
!!! [0206 15:57:19]  1: /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:798 kube::golang::build_some_binaries(...)
!!! [0206 15:57:19]  2: /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:949 kube::golang::build_binaries_for_platform(...)
!!! [0206 15:57:19]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0206 15:57:19] Error in /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:747. Exit status: 1
!!! [0206 15:57:19] Call tree:
!!! [0206 15:57:19]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0206 15:57:19] Error in /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:950. Exit status: 1
!!! [0206 15:57:19] Call tree:
!!! [0206 15:57:19]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0206 15:57:19] Error in /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:953. Exit status: 1
make: *** [Makefile:96: all] Error 1
```

and without the change:
```
$ make WHAT=cmd/kubelet
go version go1.22rc2 linux/amd64
+++ [0206 16:01:33] Building go targets for linux/amd64
    k8s.io/kubernetes/cmd/kubelet (non-static)
# k8s.io/kubernetes/pkg/kubelet/cm/dra
pkg/kubelet/cm/dra/manager.go:67:53: syntax error: unexpected error at end of statement
!!! [0206 16:01:34] Call tree:
!!! [0206 16:01:34]  1: /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:798 kube::golang::build_some_binaries(...)
!!! [0206 16:01:34]  2: /home/ed/go/src/k8s.io/kubernetes/hack/lib/golang.sh:949 kube::golang::build_binaries_for_platform(...)
!!! [0206 16:01:34]  3: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0206 16:01:34] Call tree:
!!! [0206 16:01:34]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [0206 16:01:34] Call tree:
!!! [0206 16:01:34]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make: *** [Makefile:96: all] Error 1
```

--- a/hack/lib/logging.sh
+++ b/hack/lib/logging.sh
@@ -82,10 +82,6 @@ kube::log::error_exit() {
   local stack_skip="${3:-0}"
   stack_skip=$((stack_skip + 1))
 
-  # Always print the error message, regardless of KUBE_VERBOSE value
-  kube::log::error "${message}"
-
-  # Additional debug information printed if KUBE_VERBOSE is 4 or greater
   if [[ ${KUBE_VERBOSE} -ge 4 ]]; then
     local source_file=${BASH_SOURCE[${stack_skip}]}
     local source_line=${BASH_LINENO[$((stack_skip - 1))]}

